### ISSUE TITLE
navsat_transform should wait for valid time

### ISSUE THREAD

**Issue by agoluberg** [cra-ros-pkg/robot_localization#399]
Filter node is waiting for the valid time (`ros::Time::waitForValid()`). The navsat_transform should use the same mechanism. When using the node in simulator in paused mode, the delay parameter for example is used up before I can start the simulator and therefore there is no actual delay compared to other nodes.

**Comment by ayrton04** [comment]
Sounds reasonable! Care to submit a PR?

**Comment by ayrton04** [comment]
Would you happen to have time to test [this branch](https://github.com/locusrobotics/robot_localization/tree/399-ntn-valid-time)?

**Comment by agoluberg** [comment]
Sorry, I actually created the same modification but didn't submit it here. We are using the exact same code and it seems to work just fine. I don't have a possibility to test exactly this branch. We start Gazebo in pause mode and the node now waits until unpause is initiated.

**Comment by ayrton04** [comment]
Did you [also remove the ros::Time::init()](https://github.com/cra-ros-pkg/robot_localization/compare/cra-ros-pkg:f0b540b...locusrobotics:de60fdb#diff-dc270f64cc0f5f1b532a944e353b01e0L79)?

**Comment by agoluberg** [comment]
Yes. But if I think about that, I don't remember why exactly. I checked that EKF nodes both use the same code.  With init, the `g_use_sim_time` will be set to false. But that caused the problems (if I remember correctly). This is a problem only if used with simulated time. Live systems work well without this `waitForValid()` call.

**Comment by ayrton04** [comment]
The `ros::Time::init` shouldn't be necessary, because we're calling `ros::init` in `navsat_transform_node.cpp`. Just making sure it's working for you.

**Comment by agoluberg** [comment]
Yes, of course, init is done in the node.cpp. And yes, it is working as expected. Thanks!
